### PR TITLE
Testing plan for location verification

### LINKED ISSUE #85: Location Verification Implementation Guidelines
**Problem description**
There is no clear guideline about how to implement the response to location-verification, when result is PARTIAL, and it is not obvious how to calculate the matchRate 

**Expected action**
Trigger discussion and agree on some guidelines for common implementation.

An initial proposal is presented, to cover several scenarios that may happen. Also with comments and concerns.

**Additional context**

Word Document:
[Location Verification Implementation Guidelines.docx](https://github.com/camaraproject/DeviceLocation/files/12213109/Location.Verification.Implementation.Guidelines.docx)

### LINKED ISSUE #208: Test plan for Location Verification API
**Problem description**
Test plan for Location Verification API

**Expected action**
Provide test plan aligned with [Testing guidelines in Commonalities](https://github.com/camaraproject/Commonalities/blob/main/documentation/API-Testing-Guidelines.md)

### PR DESCRIPTION
**Pull request by jlurien**
#### What type of PR is this?

* tests

#### What this PR does / why we need it:

Example of testing plan aligned with the guidelines being proposed in https://github.com/camaraproject/Commonalities/issues/158.

#### Which issue(s) this PR fixes:

Fixes #208 

#### Special notes for reviewers:

There are many comments embedded in the test plan to clarify and explain some scenarios. In some cases alternate syntax for the same functionality is used to illustrate the possibilities.

Some points are still open to discussion, such as the degree of specification of the testing plan.

### CODE REVIEW

**Review by bigludo7 — DISMISSED** [review]
Checked with @patrice-conil and work for Orange side.
Could be very helpful to provide an extract from that to answer https://github.com/camaraproject/Commonalities/issues/127 in commonalities.

**Comment by jlurien** [comment]
> Checked with @patrice-conil and work for Orange side. Could be very helpful to provide an extract from that to answer [camaraproject/Commonalities#127](https://github.com/camaraproject/Commonalities/issues/127) in commonalities.

Yes, I have to think how to align this with the work in Commonalities and the testing guidelines enhancement. I will be preparing some more detailed proposal towards the dedicated workshop.

**Comment by jlurien** [comment]
> @jlurien Accordingly to your proposal in Commonalities, this file name should location-verification.feature I guess.

True, I have to update and align my own example with my own proposal :)

**Review by bigludo7 — DISMISSED** [review]
LGTM for me
2 very minor typing errors

File: code/Test_definitions/location-verification.feature [hunk 1]
```diff
@@ -0,0 +1,264 @@
+Feature: CAMARA Device location verification API, vwip - Operation verifyLocation
+  # Input to be provided by the imnplementation to the tester
```
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:2]:
```suggestion
  # Input to be provided by the implementation to the tester

```

File: code/Test_definitions/location-verification.feature [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+  # Input to be provided by the implementation to the tester
+
+  #
+  # Implementation indications:
+  # * List of device identifier types which are not supported, among: phoneNumber, networkAccessIdentifier, ipv4Address, ipv6Address
+  #
+  # Testing assets:
+  # * A device object which location is known by the network when connected. To test all scenarios, at least 2 valid devices are needed
+  # * The known location of the testing devices
+  # * A device object identifying a device commercialized by the implemenation for which the service is not applicable, if any
```
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:11]:
```suggestion
  # * A device object identifying a device commercialized by the implementation for which the service is not applicable, if any

```

File: code/Test_definitions/location-verification.feature [hunk 3]
```diff
@@ -0,0 +1,265 @@
+Feature: CAMARA Device location verification API, vwip - Operation verifyLocation
+  # Input to be provided by the implementation to the tester
+
```
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:3]:
```suggestion
```

File: code/Test_definitions/location-verification.feature [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+  @location_verification_15_device_token_mismatch
+  Scenario: Inconsistent access token context for the device
+    # To test this, a token have to be obtained for a different device
+    Given the request body property "$.device" is set to a valid testing device
+    And the header "Authorization" is set to a valid access token emitted for a different device
+    When the HTTP "POST" request is sent
+    Then the response status code is 403
+    And the response property "$.status" is 403
+    And the response property "$.code" is "DEVICE_INVALID_TOKEN_CONTEXT"
```
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:161]:
```suggestion
    And the response property "$.code" is "INVALID_TOKEN_CONTEXT"
```

File: code/Test_definitions/location-verification.feature [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    And the response property "$.message" contains a user friendly text
+
+  @location_verification_401.2_expired_access_token
+  Scenario: Expired access token
+    Given the header "Authorization" is set to an expired access token
+    And the request body is set to a valid request body
+    When the HTTP "POST" request is sent
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "UNAUTHENTICATED"
```
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:252]:
```suggestion
    And the response property "$.code" is "AUTHENTICATION_REQUIRED"
```

File: code/Test_definitions/location-verification.feature [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+    And the response property "$.message" contains a user friendly text
+
+  @location_verification_401.2_expired_access_token
+  Scenario: Expired access token
+    Given the header "Authorization" is set to an expired access token
+    And the request body is set to a valid request body
+    When the HTTP "POST" request is sent
+    Then the response status code is 401
+    And the response property "$.status" is 401
+    And the response property "$.code" is "AUTHENTICATION_REQUIRED"
```
Comment (jlurien) [inline code/Test_definitions/location-verification.feature:252]:
```suggestion
    And the response property "$.code" is "UNAUTHENTICATED"
```

**Comment by jlurien** [comment]
@bigludo7 Wee need another explicit approval after the last fixes to merge the PR. Thanks

**Review by bigludo7 — APPROVED** [review]
LGTM

**Pull request merged**